### PR TITLE
add TokenFileAUth which reloads token if it expires

### LINKED ISSUE #293: client should periodically reload InClusterConfig token
See https://github.com/kubernetes/kubernetes/issues/70679

### PR DESCRIPTION
**Pull request by zshihang**
fixes #293. by default, this will set an expiration period of 1 minutes on service account token. as we are graduating the Beta feature Projected Service Account Volume to GA, we expect clients to reload token before it expires.

### CODE REVIEW

File: src/KubernetesClient/Authentication/TokenFileAuth.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        {
+            _token_file = token_file;
+        }
+
+        public async Task<AuthenticationHeaderValue> GetAuthenticationHeaderAsync(CancellationToken cancellationToken)
+        {
+            if (_token_expires_at < DateTime.UtcNow)
+            {
+                _token = File.ReadAllText(_token_file).Trim();
+                _token_expires_at = DateTime.UtcNow.AddSeconds(60);
```
Comment (brendandburns) [inline src/KubernetesClient/Authentication/TokenFileAuth.cs:26]: Please add a comment about why 60 seconds is the right value.

File: src/KubernetesClient/Authentication/TokenFileAuth.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Net.Http.Headers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Rest;
+
+namespace k8s.Authentication
+{
+    public class TokenFileAuth : ITokenProvider
```
Comment (brendandburns) [inline src/KubernetesClient/Authentication/TokenFileAuth.cs:10]: Please add a unit test for this file.

Comment (brendandburns) [reply src/KubernetesClient/Authentication/TokenFileAuth.cs:10]: oh, you already did, nevermind.

File: tests/KubernetesClient.Tests/TokenFileAuth.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using System.Threading.Tasks;
+using FluentAssertions;
+using k8s.Authentication;
+using Xunit;
+
+namespace k8s.Tests
+{
+    public class TokenFileAuthTests
+    {
+        [OperatingSystemDependentFact(Exclude = OperatingSystem.OSX | OperatingSystem.Windows)]
```
Comment (brendandburns) [inline tests/KubernetesClient.Tests/TokenFileAuth.cs:12]: Why is this excluded on OSX or Windows? This test should definitely work on all platforms.

**Comment by brendandburns** [comment]
Couple of small comments.

File: src/KubernetesClient/Authentication/TokenFileAuth.cs (hunk 1 above)
Comment (zshihang) [reply src/KubernetesClient/Authentication/TokenFileAuth.cs]: done

File: tests/KubernetesClient.Tests/TokenFileAuth.cs (hunk 3 above)
Comment (zshihang) [reply tests/KubernetesClient.Tests/TokenFileAuth.cs]: removed. i only had linux at hand to verify the test.

**Pull request merged**